### PR TITLE
Decompile Various Weapon Functions

### PR DESCRIPTION
**Pull request by hohle**
* `w_006` - `func_ptr_80170008`
* `w_016` - `func_ptr_80170010`
* `w_022` - `EntityWeaponAttack`

### CODE REVIEW

**Comment by hohle** [comment]
Checking failure.

File: src/weapon/w_016.c [hunk 1]
```diff
@@ -179,7 +179,42 @@ void func_ptr_8017000C(Entity* self) {
...
+extern s32 D_74000_8017BD74;
+
+s32 func_ptr_80170010(Entity* self) {
+    Entity* factory;
+
+    if (self->step == 0) {
+        self->animSet = 2;
+        self->unk4C = D_74000_8017A5B0;
+        self->zPriority = PLAYER.zPriority - 4;
+        self->flags = 0x08100000;
```
Comment (sozud) [inline src/weapon/w_016.c:192]: flags enum

File: src/weapon/w_016.c [hunk 2]
```diff
@@ -179,7 +179,42 @@ void func_ptr_8017000C(Entity* self) {
...
+                } else {
+                    factory->entityId = 0xFF;
+                }
+            }
+        }
+        self->posY.i.hi += -6 + (rand() % 12);
+        self->posX.i.hi += -3 + (rand() & 7);
+        D_74000_8017BD74 += 1;
+        if (!(rand() & 1)) {
+            self->drawMode |= 0x70;
```
Comment (sozud) [inline src/weapon/w_016.c:209]: drawMode enum

File: src/weapon/w_022.c [hunk 3]
```diff
@@ -6,7 +6,61 @@
...
+    case 1:
+        if (self->animFrameDuration < 0) {
+            self->ext.timer.t = 16;
+            self->drawFlags |= DRAW_MENU;
+            self->step++;
+        }
+        if (self->animFrameIdx == 20 && self->animFrameDuration == 2 &&
+            g_api.CreateEntFactoryFromEntity(
+                self, ((g_HandId + 1) << 0xC) | 0x38, 0)) {
+            g_api.PlaySfx(0x625);
```
Comment (sozud) [inline src/weapon/w_022.c:52]: I think this is in the sfx enum?

**Pull request merged**